_plugins/rosindex_generator.rb

```diff
@@ -666,30 +673,38 @@ def generate_sorted_paginated(site, elements_sorted, default_sort_key, n_element
     end
   end
 
-
   def sort_repos(site)
     repos_sorted = {}
     repos_sorted['name']= @repo_names.sort_by { |name, instances| name }
 
-    repos_sorted['time'] = repos_sorted['name'].reverse.sort_by { |name, instances|
-      instances.default.snapshots.reject { |d,s|
-        not $recent_distros.include?(d)
-      }.map { |d,s|
-        s.data['last_commit_time'].to_s
-      }.max
-    }.reverse
+    repos_sorted['time'] = Hash.new
+    $all_distros.each do |distro|
+      repos_sorted['time'][distro] = repos_sorted['name'].reverse.sort_by { |name, instances|
+        instances.default.snapshots.reject { |d,s|
+          s.nil? or not $recent_distros.include?(d) or not distro == d
```

Comment:
@apojomovsky consider `select`ing `instances.default.snapshots` based on distro instead for better readability.